src/BlockTypes/HandpickedProducts.php

```diff
@@ -32,6 +32,30 @@ protected function set_block_query_args( &$query_args ) {
 		$query_args['posts_per_page'] = count( $ids );
 	}
 
+	/**
+	 * Set visibility query args. Handpicked products will show hidden products if chosen.
+	 *
+	 * @param array $query_args Query args.
+	 */
+	protected function set_visibility_query_args( &$query_args ) {
+		$product_visibility_terms  = wc_get_product_visibility_term_ids();
+		$product_visibility_not_in = array();
+
+		// Hide out of stock products.
+		if ( 'yes' === get_option( 'woocommerce_hide_out_of_stock_items' ) ) {
+			$product_visibility_not_in[] = $product_visibility_terms['outofstock'];
+		}
+
+		if ( ! empty( $product_visibility_not_in ) ) {
```

Comment:
This probably could be simplified because the only time there will be `$product_visibility_not_in` will be if there are out of stock items.  So maybe something like:

```php
/**
 * Set visibility query args.
 *
 * @param array $query_args Query args.
 */
protected function set_visibility_query_args( &$query_args ) {
	if ( 'yes' === get_option( 'woocommerce_hide_out_of_stick_items' ) ) {
		$product_visibility_terms  = wc_get_product_visibility_term_ids();
		$query_args['tax_query'][] = array(
			'taxonomy' => 'product_visibility',
			'field'    => 'term_taxonomy_id',
			'terms'    => array( $product_visibility_terms['outofstock'] ),
			'operator' => 'NOT IN ',
		);
	}
}
```